### PR TITLE
feature/edit/subset

### PR DESCRIPTION
**Pull request by benrr101**
This change adds the edit/subset command that returns a subset of rows with pending edits applied to it.

This change also adds some very important bug fixes to the LongList class that basically allow it to work properly.

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/CellUpdate.cs [hunk 1]
```diff
@@ -86,6 +86,22 @@ public CellUpdate(DbColumnWrapper column, string valueAsString)
         #region Properties
 
         /// <summary>
+        /// Converts the cell update to a DbCellValue
+        /// </summary>
+        public DbCellValue AsDbCellValue
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/CellUpdate.cs]: the naming of this property suggests it should probably be a method (i.e. "As" is a verb)

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs [hunk 2]
```diff
@@ -422,6 +392,28 @@ public void InitializeService(ServiceHost serviceHost)
             }
         }
 
+        /// <summary>
+        /// Retrieves the requested subset of rows from the requested result set. Intended to
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs]: >Intended to [](start = 82, length = 11)

comment seems to cut off

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs [hunk 3]
```diff
@@ -422,6 +392,28 @@ public void InitializeService(ServiceHost serviceHost)
...
+        /// </summary>
+        /// <param name="subsetParams">Parameters for the subset to retrieve</param>
+        /// <returns>The requested subset</returns>
+        /// <exception cref="ArgumentOutOfRangeException">The requested query does not exist</exception>
+        public async Task<ResultSetSubset> InterServiceResultSubset(SubsetParams subsetParams)
```
Comment (kburtram) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs]: >InterService [](start = 43, length = 12)

I'm not quite clear what "InterService" means in this context.  Perhaps a different name (or completed comments :) ) may make it more intuitive.

**Comment by kburtram** [comment]
:shipit:

File: src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/CellUpdate.cs (hunk 1 above)
Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/EditData/UpdateManagement/CellUpdate.cs]: Not to sound grammar obsessed, but in this case "as" is used as a preposition :)
As such, I think it works fine as-is. (Yes, pun intended)

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs (hunk 2 above)
Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs]: I have a tendency to do this. Will fix.

File: src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs (hunk 3 above)
Comment (benrr101) [inline src/Microsoft.SqlTools.ServiceLayer/QueryExecution/QueryExecutionService.cs]: That was the bit of the comment that was left off, lol...

Basically I picked the name "interservice" to indicate that it's meant to be called by another service (in this case the EditDataService). This bypasses the request context and implementation that assumes its being called by the JSON RPC. We also use the same methods from the JSON RPC request handlers, so I'm trying to keep the implementations here DRY.

**Pull request merged**